deps/chakrashim/core/bin/ch/WScriptJsrt.cpp

```diff
@@ -0,0 +1,596 @@
+        const wchar_t *fileName;
+        const wchar_t *scriptInjectType = L"self";
+        size_t fileNameLength;
+        size_t scriptInjectTypeLength;
+
+        IfJsrtErrorSetGo(ChakraRTInterface::JsStringToPointer(arguments[1], &fileName, &fileNameLength));
+
+        if (argumentCount > 2)
+        {
+            IfJsrtErrorSetGo(ChakraRTInterface::JsStringToPointer(arguments[2], &scriptInjectType, &scriptInjectTypeLength));
+        }
+
+        if (errorCode == JsNoError)
+        {
+            hr = Helpers::LoadScriptFromFile(fileName, fileContent);
+            if (FAILED(hr))
+            {
+                fwprintf(stderr, L"Couldn't load file.\n");
+            }
+            else
+            {
+                returnValue = LoadScript(callee, fileName, fileNameLength, fileContent, scriptInjectType);
+            }
+        }
+    }
+
+Error:
+    if (errorCode != JsNoError)
+    {
+        JsValueRef errorObject;
+        JsValueRef errorMessageString;
+
+        if (wcscmp(errorMessage, L"") == 0) {
+            errorMessage = ConvertErrorCodeToMessage(errorCode);
+        }
+
+        ChakraRTInterface::JsPointerToString(errorMessage, wcslen(errorMessage), &errorMessageString);
+        ChakraRTInterface::JsCreateError(errorMessageString, &errorObject);
+        ChakraRTInterface::JsSetException(errorObject);
+    }
+
+    return returnValue;
+}
+
+JsValueRef __stdcall WScriptJsrt::LoadScriptCallback(JsValueRef callee, bool isConstructCall, JsValueRef *arguments, unsigned short argumentCount, void *callbackState)
+{
+    HRESULT hr = E_FAIL;
+    JsErrorCode errorCode = JsNoError;
+    LPCWSTR errorMessage = L"";
+    JsValueRef returnValue = JS_INVALID_REFERENCE;
+
+    if (argumentCount < 2 || argumentCount > 4)
+    {
+        errorCode = JsErrorInvalidArgument;
+        errorMessage = L"Need more or fewer arguments for WScript.LoadScript";
+    }
+    else
+    {
+        const wchar_t *fileContent;
+        const wchar_t *fileName;
+        const wchar_t *scriptInjectType = L"self";
+        size_t fileContentLength;
+        size_t fileNameLength;
+        size_t scriptInjectTypeLength;
+
+        IfJsrtErrorSetGo(ChakraRTInterface::JsStringToPointer(arguments[1], &fileContent, &fileContentLength));
+
+        if (argumentCount > 2)
+        {
+            IfJsrtErrorSetGo(ChakraRTInterface::JsStringToPointer(arguments[2], &scriptInjectType, &scriptInjectTypeLength));
+        }
+
+        fileName = L"script.js";
+        fileNameLength = wcslen(fileName);
+        if (argumentCount > 3)
+        {
+            IfJsrtErrorSetGo(ChakraRTInterface::JsStringToPointer(arguments[3], &fileName, &fileNameLength));
+        }
+        returnValue = LoadScript(callee, fileName, fileNameLength, fileContent, scriptInjectType);
+    }
+
+Error:
+    if (errorCode != JsNoError)
+    {
+        JsValueRef errorObject;
+        JsValueRef errorMessageString;
+
+        if (wcscmp(errorMessage, L"") == 0) {
```

Comment:
@nathanpeck Please point comments about Chakracore files to https://github.com/Microsoft/ChakraCore -- there's not much we are going to do about it here. :)